### ISSUE TITLE
Adding Runner: Segmentation fault "$DIR"/bin/Runner.Listener run $* after running ./run.sh

### ISSUE THREAD

**Issue by kktesting-admin** [actions/runner#632]
Hi,

I'm trying to configure a self-hosted runner on a new VPS running Linux 18.04 64bit. I just followed the Download and Configure steps, but I keep on encountering the following after running ./run.sh.

![image](https://user-images.githubusercontent.com/68962479/89091432-4d37f480-d3dc-11ea-804c-f32230f59a29.png)

I ran ./run.sh a few times and looks like it is running the listener fine, but once I push changes on my repository the build fails with the same error.

**Comment by ethanchewy** [comment]
Just curious, did this occur after the runner updated?

**Comment by nikola-jokic** [comment]
Hi @kktesting-admin,

Does the machine have enough resources to run this? Since the Segmentation fault is occurring, and it fails to run jobs with the same error, it makes me wonder if the machine contains enough resources to run those tasks? Could you please double check if that is causing your problem?

**Comment by nikola-jokic** [comment]
I'm going to close out this issue until we hear back from you, please let us know if you are still seeing this issue!

**Comment by nickflees** [comment]
I am having the same problem. Immediately after running config successfully, an attempt to run ./run.sh results in a segmentation fault:
`/opt/actions-runner/run-helper.sh: line 36: 45369 Segmentation fault      (core dumped) "$DIR"/bin/Runner.Listener run $*`

I observed this behavior for the first time this morning with a runner that has run without issue for months. Based on some cursory troubleshooting, I suspected that system resources were a problem, so I increased available disk space and confirmed that there is a reasonable amount of RAM available.

I'm attempting to run the Actions runner on an Ubuntu Linux 22.04 VM on Azure.

Here's the tail end of the runner log:

```
[2024-02-06 22:50:27Z INFO MessageListener] Loading Credentials
[2024-02-06 22:50:27Z INFO ConfigurationStore] HasCredentials()
[2024-02-06 22:50:27Z INFO ConfigurationStore] stored True
[2024-02-06 22:50:27Z INFO CredentialManager] GetCredentialProvider
[2024-02-06 22:50:27Z INFO CredentialManager] Creating type OAuth
[2024-02-06 22:50:27Z INFO CredentialManager] Creating credential type: OAuth
[2024-02-06 22:50:27Z INFO HostContext] Well known directory 'Bin': '/opt/actions-runner/bin'
[2024-02-06 22:50:27Z INFO HostContext] Well known directory 'Root': '/opt/actions-runner'
[2024-02-06 22:50:27Z INFO HostContext] Well known config file 'RSACredentials': '/opt/actions-runner/.credentials_rsaparams'
[2024-02-06 22:50:27Z INFO RSAFileKeyManager] Loading RSA key parameters from file /opt/actions-runner/.credentials_rsaparams
[2024-02-06 22:50:27Z INFO MessageListener] Attempt to create session.
[2024-02-06 22:50:27Z INFO MessageListener] Connecting to the Runner Server...
[2024-02-06 22:50:27Z INFO RunnerServer] EstablishVssConnection
[2024-02-06 22:50:27Z INFO RunnerServer] Establish connection with 100 seconds timeout.
[2024-02-06 22:50:28Z INFO GitHubActionsService] Starting operation Location.GetConnectionData
[2024-02-06 22:50:28Z INFO RunnerServer] EstablishVssConnection
[2024-02-06 22:50:28Z INFO RunnerServer] Establish connection with 60 seconds timeout.
[2024-02-06 22:50:28Z INFO GitHubActionsService] Starting operation Location.GetConnectionData
[2024-02-06 22:50:28Z INFO RunnerServer] EstablishVssConnection
[2024-02-06 22:50:28Z INFO RunnerServer] Establish connection with 60 seconds timeout.
[2024-02-06 22:50:28Z INFO GitHubActionsService] Starting operation Location.GetConnectionData
```

**Comment by nickflees** [comment]
Some additional information:

As a clarification, I had a runner on this VM for months without any issues. This morning, I noticed that GitHub had lost connectivity and found that the runner had been seg-faulting repeatedly throughout the night. Having failed to resolve the issue by rebooting the VM and trying to run the runner again, I deleted the runner and started over. This is why I ran the config immediately prior to the segmentation fault I reported above.

- I'm running v2.312.0
- The VM has 16 GB of RAM and not much running on it
- The VM has a data disk of 512 GB, of which about 95% is free
